include/xgboost/json.h

```diff
@@ -256,15 +262,16 @@ class JsonBoolean : public Value {
 };
 
 struct StringView {
```

Comment:
Aside: we may want to contribute this class into dmlc-core (`dmlc::StringView`)

Reply:
@hcho3 There are quite a lot of requirements to make it really useful.  One from them is `{from, to}_chars` from c++17.  I tried to adopt it from `libcxx` from llvm project, but a few thousands lines of code is unavoidable.